modules/3d/src/pointcloud/io_obj.cpp

```diff
@@ -98,10 +177,53 @@ void ObjDecoder::readData(std::vector<Point3f> &points, std::vector<Point3f> &no
         }
     }
 
+    if (duplicateVertices)
+    {
+        points.clear();
+        normals.clear();
+        rgb.clear();
+        texCoords.clear();
+        indices.clear();
+
+        for (int tri = 0; tri < (int)idxList.size(); tri++)
+        {
+            auto vi = idxList[tri];
+            auto ti = texIdxList[tri];
+            auto ni = normalIdxList[tri];
+
+            std::vector<int32_t> newvi;
+            newvi.reserve(3);
+            for (int i = 0; i < (int)vi.size(); i++)
+            {
+                newvi.push_back((int)points.size());
+                points.push_back(ptsList.at(vi[i]));
+                if (!rgbList.empty())
+                {
+                    rgb.push_back(rgbList.at(vi[i]));
+                }
+
+                texCoords.push_back(texCoordList.at(ti[i]));
+                normals.push_back(nrmList.at(ni[i]));
```

Comment:
It makes sense to call reserve() for all vectors, to prevent reallocs on push_back().